### ISSUE TITLE
vic-machine support resources in folder

### ISSUE THREAD

**Issue by emlin** [vmware/vic#3817]
#For stories, please include the information below:

**User Statement:**

As a vic admin, I want to user vic-machine to create and manage VCH endpoint VM with resources under VC folders. 

**Details:**
following changes for vic-machine create

1.    --target option will have dc folder specified, e.g. "vc ip"/folder/dc, and it's also possible to be "vc ip"/dc (even the dc is under folder), and also "vc ip"/folder valid, in which case, there is only one dc in that folder.
2.    --compute-resource, with same reason, it will be folder/cluster, folder/rp (in case there is only one cluster in that folder), cluster, or cluster/rp, four possible format.
3.    The storage options. It already have a format as datastore/path, and it's ok to leave path empty, but this change, it will vary as well. User might specify folder/datastore/path, folder/datastore, or datastore/path.
4.    network options are usually only name specified, that is easy in this change.
5.    resource suggestion includes folders

Other vic-machine commands should all support search with name through govmomi

**Acceptance Criteria:**
Following test should be included:
- Specify resources with all supported formats, to create VCH
- Wrong resource name will get correct resource suggestion for resources under folder

**Comment by emlin** [comment]
Depends on #3619, so set it to low priority

**Comment by dougm** [comment]
Trying this with #3620 merged:

- Need a small change to support datacenter name in target without the folder path

- We should be able to use a host or cluster by name, without the full path for --compute-resource, but is currently hardcoded to append "Resources" if path is not absolute

- Datastore suggestion/validation works as-is when datastore lives in a folder and just the name is given

- Network suggestion/validation works as-is when network lives in a folder and just the name is given

- Resource suggestion works when Host/Cluster is inside a folder (but currently needs the full path to validate as mentioned above)

**Comment by stuclem** [comment]
@mdubya66 does this really need to go in the release notes?

**Comment by stuclem** [comment]
Removing kind/note

**Comment by hickeng** [comment]
@dougm please add references to the commits that addressed this.

**Comment by dougm** [comment]
The remaining issues here were fixed by the changes for #4203
